From 36789208ed899a2192cfb3a98c83927a9d5d7267 Mon Sep 17 00:00:00 2001
From: Dongxu Sun <sundongxu3@huawei.com>
Date: Thu, 11 Jan 2024 20:43:19 +0800
Subject: [PATCH 524/960] downstream: mbigen: vtimer: isolate mbigen vtimer
 funcs with macro

use CONFIG_HISILICON_IRQ_MBIGEN to isolate mbigen vtimer
funcs in ./include/clocksource/arm_arch_timer.h file.

Signed-off-by: Dongxu Sun <sundongxu3@huawei.com>
Signed-off-by: Slim6882 <15605196882@163.com>
Signed-off-by: Qi Xi <xiqi2@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 arch/arm64/kvm/Kconfig | 5 +++++
 config.aarch64         | 1 +
 config.aarch64-64k     | 1 +
 3 files changed, 7 insertions(+)

diff --git a/arch/arm64/kvm/Kconfig b/arch/arm64/kvm/Kconfig
index 8350d43f56d4..f5a9852d26fa 100644
--- a/arch/arm64/kvm/Kconfig
+++ b/arch/arm64/kvm/Kconfig
@@ -72,4 +72,9 @@ config PROTECTED_NVHE_STACKTRACE
 
 	  If unsure, or not using protected nVHE (pKVM), say N.
 
+config VIRT_VTIMER_IRQ_BYPASS
+	bool "MBIGEN enable vtimer irq bypass"
+	depends on KVM && ARM64 && ARCH_HISI && HISILICON_IRQ_MBIGEN
+	default n
+
 endif # VIRTUALIZATION
diff --git a/config.aarch64 b/config.aarch64
index d8772a8e3f09..fe2708280ae7 100644
--- a/config.aarch64
+++ b/config.aarch64
@@ -412,6 +412,7 @@ CONFIG_ARM64_HAFT=y
 CONFIG_ARCH_HAS_NONLEAF_PMD_YOUNG=y
 CONFIG_KVM_HISI_VIRT=y
 CONFIG_VIRT_PLAT_DEV=y
+CONFIG_VIRT_VTIMER_IRQ_BYPASS=y
 
 CONFIG_ARM64_4K_PAGES=y
 # CONFIG_ARM64_16K_PAGES is not set
diff --git a/config.aarch64-64k b/config.aarch64-64k
index 753e9d93528b..1d812e568933 100644
--- a/config.aarch64-64k
+++ b/config.aarch64-64k
@@ -412,6 +412,7 @@ CONFIG_ARM64_HAFT=y
 CONFIG_ARCH_HAS_NONLEAF_PMD_YOUNG=y
 CONFIG_KVM_HISI_VIRT=y
 CONFIG_VIRT_PLAT_DEV=y
+CONFIG_VIRT_VTIMER_IRQ_BYPASS=y
 
 # CONFIG_ARM64_4K_PAGES is not set
 # CONFIG_ARM64_16K_PAGES is not set
-- 
2.43.0

